.travis.yml

```diff
@@ -80,6 +80,34 @@ jobs:
         - python2 ./scripts/githooks/commit_msg_tests.py
         - python3 ./scripts/githooks/commit_msg_tests.py
 
+        # This runs tests on ARM32 emulation
+    - if: (type = pull_request) OR (type = cron) OR NOT (branch = master) # run on PRs for debugging todo:: change to pre-release before merge
+      stage: test
+      name: "Regression 9.6 test ARM processors"
+      env:
+        # - PG_VERSION=10.2 PG_GIT_TAG=REL_10_2 ADDITIONAL_CMAKE_FLAGS="-DUSE_DEFAULT_VISIBILITY=1"
+      before_install:
+        - docker run --rm --privileged multiarch/qemu-user-static:register --reset
+        - docker run -d --name docker_arm_emulator -v ${TRAVIS_BUILD_DIR}:/timescaledb ngalstyan4/timescaledb-arm:arm9.6 /bin/sleep infinity
+        - docker exec docker_arm_emulator /bin/bash -c "chown -R postgres:postgres /timescaledb"
+      install:
+        - docker exec docker_arm_emulator /bin/bash -c "mkdir /build && chown postgres:postgres /build"
+        # todo:: add CFLAGS='-Werror -O2'  prefix to cmake when timestamp difference is fixed.
+      #  - docker exec -it docker_arm_emulator /bin/sh -c "cd /tsdb_build && cmake /timescaledb -DCMAKE_BUILD_TYPE=Debug -DPG_SOURCE_DIR=/postgresqlsrc ${OTHER_CMAKE_FLAGS} && make install"
+        - docker exec -u postgres docker_arm_emulator /bin/bash -c "cd /build && cmake /timescaledb -DCMAKE_BUILD_TYPE=Debug && make"
+        - docker exec docker_arm_emulator /bin/bash -c "cd /build && make install"
+
+      before_script:
+      after_failure:
+      # Q:: in an earlier place timescaledb puts tests under /postgres/timescaledb/build/debug/ but here there is no debug, why?
+        - docker exec -u postgres -it docker_arm_emulator cat /build/tsl/test/log/initdb.log /timescaledb/build/test/regression.diffs /timescaledb/build/tsl/test/regression.diffs /timescaledb/build/test/isolation/regression.diffs /timescaledb/build/test/pgtest/regression.diffs && echo $LANG $LC_ALL
+      script:
+      # allow 50 mins to run
+        - travis_wait 50 docker exec -u postgres docker_arm_emulator /bin/bash -c "cd /build && make installcheck IGNORES='multi_transaction_indexing sql_query_results_unoptimized sql_query_results_x_diff chunk_utils bgw_db_scheduler continuous_aggs_insert plan_expand_hypertable-9.6 sql_query_results_unoptimized-9.6 sql_query_results_x_diff-9.6 plan_hashagg-9.6'"
```

Comment:
travis_wait is not tty so `docker exec -it` will not work which means one cannot see continuous output in travis from this command. A workaround is described [here](https://github.com/travis-ci/travis-ci/issues/6934) but this approach would mean tests are marked as completed exactly after the specified wait time passes and not when the testing finishes